compiler/src/compile.rs

```diff
@@ -375,26 +383,39 @@ impl Compiler {
         Ok(())
     }
 
-    fn load_name(&mut self, name: &str) {
+    fn load_name(&mut self, name: &str) -> CompileResult<()> {
         self.compile_name(name, NameUsage::Load)
     }
 
     fn store_name(&mut self, name: &str) -> CompileResult<()> {
-        if name.eq("__debug__") {
-            return Err(self.error(CompileErrorType::SyntaxError(
-                "cannot assign to __debug__".to_owned(),
-            )));
-        }
-        self.compile_name(name, NameUsage::Store);
-        Ok(())
+        self.compile_name(name, NameUsage::Store)
     }
 
     fn mangle<'a>(&self, name: &'a str) -> Cow<'a, str> {
         symboltable::mangle_name(self.class_name.as_deref(), name)
     }
 
-    fn compile_name(&mut self, name: &str, usage: NameUsage) {
+    fn check_forbidden_name(&self, name: &str, usage: NameUsage) -> CompileResult<()> {
+        if usage == NameUsage::Store && is_forbidden_name(name) {
+            return Err(self.error(CompileErrorType::SyntaxError(format!(
+                "cannot assign to {}",
+                name
+            ))));
+        }
+        if usage == NameUsage::Delete && is_forbidden_name(name) {
+            return Err(self.error(CompileErrorType::SyntaxError(format!(
+                "cannot delete {}",
+                name
+            ))));
+        }
+        Ok(())
```

Comment:
```suggestion
        if usage == NameUsage::Store && is_forbidden_name(name) {
            return Err(self.error(CompileErrorType::SyntaxError(format!(
                "cannot assign to {}",
                name
            ))));
        }
        if usage == NameUsage::Delete && is_forbidden_name(name) {
            return Err(self.error(CompileErrorType::SyntaxError(format!(
                "cannot delete {}",
                name
            ))));
        }
        Ok(())
```

Reply:
@muscar oh, I thought I changed it quickly enough but it wasn't. could you check the newer version of this comment? I am sorry to make confusion.